tools/obs/httpClientWrapper.go

```diff
@@ -0,0 +1,120 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	logger "github.com/multiversx/mx-chain-logger-go"
+)
+
+var log = logger.GetOrCreate("main")
+
+var errInvalidValue = errors.New("invalid value")
+
+const (
+	minRequestTimeoutSec = 1
+	contentTypeKey       = "Content-Type"
+	contentTypeValue     = "application/json"
+)
+
+type httpClientWrapper struct {
+	httpClient       *http.Client
+	useAuthorization bool
+	username         string
+	password         string
+	baseUrl          string
+}
+
+// HTTPClientWrapperArgs defines the arguments needed for http client creation
+type HTTPClientWrapperArgs struct {
+	UseAuthorization  bool
+	Username          string
+	Password          string
+	BaseUrl           string
+	RequestTimeoutSec int
+}
+
+// NewHTTPWrapperClient creates an instance of httpClient which is a wrapper for http.Client
+func NewHTTPWrapperClient(args HTTPClientWrapperArgs) (*httpClientWrapper, error) {
+	err := checkArgs(args)
+	if err != nil {
+		return nil, err
+	}
+
+	httpClient := &http.Client{}
+	httpClient.Timeout = time.Duration(args.RequestTimeoutSec) * time.Second
+
+	return &httpClientWrapper{
+		httpClient:       httpClient,
+		useAuthorization: args.UseAuthorization,
+		username:         args.Username,
+		password:         args.Password,
+		baseUrl:          args.BaseUrl,
+	}, nil
+}
+
+func checkArgs(args HTTPClientWrapperArgs) error {
+	if args.RequestTimeoutSec < minRequestTimeoutSec {
```

Comment:
perhaps should add check for empty BaseUrl and empty username and passwrod in case useAuthorization is true?

Reply:
added more checks